frontend/components/apps/AppCard.tsx

```diff
@@ -1,33 +1,70 @@
 import { useQuery } from '@apollo/client'
-import { FaBox, FaBoxes, FaCube, FaUser, FaUsers } from 'react-icons/fa'
+import { FaBox, FaBoxes, FaCube, FaUser, FaUsers, FaProjectDiagram } from 'react-icons/fa'
 import { Card } from '../common/Card'
+import clsx from 'clsx'
 import GetAppMembers from '@/graphql/queries/apps/getAppMembers.gql'
 import GetAppEnvironments from '@/graphql/queries/secrets/getAppEnvironments.gql'
 import { AppType } from '@/apollo/graphql'
+import { ProviderIcon } from '../syncing/ProviderIcon'
+import GetAppSyncStatus from '@/graphql/queries/syncing/getAppSyncStatus.gql'
+import React from 'react'
 
 interface AppCardProps {
   app: AppType
 }
 
-export const AppCard = (props: AppCardProps) => {
-  const { name, id: appId } = props.app
-  const { data: appMembersData } = useQuery(GetAppMembers, { variables: { appId } })
-  const { data: appEnvsData } = useQuery(GetAppEnvironments, {
-    variables: {
-      appId,
-    },
+export const AppCard: React.FC<AppCardProps> = ({ app }) => {
+  const { name, id: appId } = app;
+  //cleaned up the code look
+  const { data: appMembersData } = useQuery(GetAppMembers, { variables: { appId } });
+  const { data: appEnvsData } = useQuery(GetAppEnvironments, { variables: { appId } });
+  const { data: syncData } = useQuery(GetAppSyncStatus, { variables: { appId }, pollInterval: 10000 });
+
+
+  //storing number of syncs
+  const providerCounts = {
+    cloudflare: 0,
+    aws: 0,
+    github: 0,
+    gitlab: 0,
+    hashicorp_vault: 0,
+    hashicorp_nomad: 0,
+    others: 0,
+  };
+
+  //Counting number of syncs for each integration
+  if (syncData) {
+    syncData.syncs.forEach((item:any) => {
+      const providerName = item.authentication.name.toLowerCase();
+      if (providerName.includes('cloudflare')) providerCounts.cloudflare += 1;
+      else if (providerName.includes('aws')) providerCounts.aws += 1;
+      else if (providerName.includes('github')) providerCounts.github += 1;
+      else if (providerName.includes('gitlab')) providerCounts.gitlab += 1;
+      else if (providerName.includes('hashicorp_vault')) providerCounts.hashicorp_vault += 1;
+      else if (providerName.includes('hashicorp_nomad')) providerCounts.hashicorp_nomad += 1;
+      else providerCounts.others += 1;
+      
+    });
+  }
+  let itemCount = 0
+  let iconPrint = 1
+  const noOfIcons = Object.values(providerCounts).filter((value) => value > 0).length;
+  //Counting number of values with are greater than 0
+  Object.entries(providerCounts).map(([provider, count]) => {
+    itemCount = itemCount+count
```

Comment:
This approach is not ideal. Each new provider that is added will require updating this code, and amount of if/else logic is not the most elegant. Instead, a better approach would be using a `flatMap` to directly infer the providers from the environments data:

```js
    const providers: string[] = environments
        .flatMap((env) => {
          return env!.syncs.map((sync) => {
            const serviceInfo = sync!.serviceInfo
            const providerId = serviceInfo!.provider!.id
            return providerId
          })
        })
        .filter((id, index, array) => array.indexOf(id) === index)
    
```

The total count can be computed with `reduce`: 

```js
environments.reduce((acc, env) => acc + (env!.syncs?.length || 0), 0)
```

And the 'extra' or 'surplus' integrations (number of integrations more than 5) can be computed as:

```js
const surplusSynCount = providers.length > 5 ? providers.length - 5 : 0
```